src/main/java/edu/ie3/util/io/FileIOUtils.java

```diff
@@ -141,6 +157,495 @@ public static BufferedWriter getBufferedWriter(
+   * @param dirName path of the directory that should be compressed
+   * @param validatedOutputFile validated output file (target archive file)
+   * @return a boolean which is either true on success or false otherwise
+   * @throws FileException If unable to write to the output stream or there is some error in the
+   *     output filepath
+   */
+  private static boolean compressDir(final Path dirName, File validatedOutputFile)
+      throws FileException {
+    Path validatedOutputFileName;
+
+    try {
+      validatedOutputFileName = validatedOutputFile.toPath();
+    } catch (InvalidPathException e) {
+      throw new FileException("Invalid output file path '" + validatedOutputFile + "'.", e);
+    }
+
+    /* Open a stream and add content to the archive */
+    try (BufferedOutputStream bufferedOutputStream =
+            new BufferedOutputStream(Files.newOutputStream(validatedOutputFileName));
+        GzipCompressorOutputStream gzipOutputStream =
+            new GzipCompressorOutputStream(bufferedOutputStream);
+        TarArchiveOutputStream tarOutputStream = new TarArchiveOutputStream(gzipOutputStream)) {
+      Files.walkFileTree(
+          dirName,
+          new SimpleFileVisitor<Path>() {
+            @Override
+            public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
+                throws IOException {
+              /* Skip symbolic link */
+              if (attrs.isSymbolicLink()) return FileVisitResult.CONTINUE;
+
+              /* Copy files to archive */
+              Path relFile = dirName.relativize(file);
+              TarArchiveEntry entry = new TarArchiveEntry(file.toFile(), relFile.toString());
+              tarOutputStream.putArchiveEntry(entry);
+              Files.copy(file, tarOutputStream);
+              tarOutputStream.closeArchiveEntry();
+
+              return FileVisitResult.CONTINUE;
+            }
+
+            @Override
+            public FileVisitResult visitFileFailed(Path file, IOException exc) throws IOException {
+              logger.error(
+                  "Unable to add '{}' to '{}'. Stopping compression.",
+                  file,
+                  validatedOutputFileName,
+                  exc);
+              return FileVisitResult.TERMINATE;
+            }
+          });
+
+      /* Close everything properly */
+      tarOutputStream.finish();
+      return true;
+    } catch (IOException e) {
+      throw new FileException("Unable to write to '" + validatedOutputFileName + "'.", e);
+    }
+  }
+
+  /**
+   * Checks, if the input file path points to a valid directory and returns the corresponding File
+   * object.
+   *
+   * @param dirName path of the directory that should be compressed
+   * @return File object corresponding to the directory to be compressed
+   * @throws FileException If the input path is null or the input path is not of a valid directory
+   */
+  private static File validateInputDirName(final Path dirName) throws FileException {
```

Comment:
Please reorganize the methods of this class, so that each one appears directly after the method, where they are called first.

Reply:
Got it. I will reorganize them.